Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -954,54 +954,70 @@ bool Element::serializes_as_void() const
 }
 
 // https://drafts.csswg.org/cssom-view/#dom-element-getboundingclientrect
-GC::Ref<Geometry::DOMRect> Element::get_bounding_client_rect() const
+GC::Ref<Geometry::DOMRect> Element::get_bounding_client_rect_for_bindings() const
+{
+    auto rect = get_bounding_client_rect();
+    return MUST(Geometry::DOMRect::construct_impl(realm(), static_cast<double>(rect.x()), static_cast<double>(rect.y()), static_cast<double>(rect.width()), static_cast<double>(rect.height())));
+}
+
+// https://drafts.csswg.org/cssom-view/#dom-element-getboundingclientrect
+CSSPixelRect Element::get_bounding_client_rect() const
 {
     // 1. Let list be the result of invoking getClientRects() on element.
     auto list = get_client_rects();
 
     // 2. If the list is empty return a DOMRect object whose x, y, width and height members are zero.
-    if (list->length() == 0)
-        return Geometry::DOMRect::construct_impl(realm(), 0, 0, 0, 0).release_value_but_fixme_should_propagate_errors();
+    if (list.size() == 0)
+        return { 0, 0, 0, 0 };
 
     // 3. If all rectangles in list have zero width or height, return the first rectangle in list.
     auto all_rectangle_has_zero_width_or_height = true;
-    for (auto i = 0u; i < list->length(); ++i) {
-        auto const& rect = list->item(i);
-        if (rect->width() != 0 && rect->height() != 0) {
+    for (auto i = 0u; i < list.size(); ++i) {
+        auto const& rect = list.at(i);
+        if (rect.width() != 0 && rect.height() != 0) {
```

Comment:
`!rect.is_empty()`

Reply:
`is_empty()` also includes negative sizes though? :thonk:

Reply:
Oh fair! Not sure if it should do that :thonker: